### ISSUE TITLE
Spark waitUntilFinish() does not wait when using Duration.Inf

### ISSUE THREAD

**Issue by JozoVilcek** [spotify/scio#2085]
I have a pipeline Scio: `0.8.0-alpha2` running on `SparkRunner`. To kick off pipeline I was using
`sc.close()` with `blocking = true` in `ScioOptions`. Migrating to `0.8.0-alpha2` I am doing now 

`sc.run().waitUntilFinish()`
but it does not block and wait for pipeline to finish. On the other hand, as a workaround, I am doing

`sc.pipeline.run().waitUntilFinish()`
which behaves as expected and waits untill job is finished.

**Comment by JozoVilcek** [comment]
This does not feels right for `Duration.Inf` wait
https://github.com/spotify/scio/blob/master/scio-core/src/main/scala/com/spotify/scio/ScioContext.scala#L337

But even when specifying duration explicitly, it does not wait.

**Comment by regadas** [comment]
Hi @JozoVilcek Which version are you migrating from?

Regarding the `Duration.Inf` it seems that it's adhering to apache beam `PipelineResult`... values `<=0` wait inf. However, from a brief look it seems that `SparkRunner` does not follow it.

**Comment by JozoVilcek** [comment]
Can you share a reference to spark runner code which is casing this not wait problem?

**Comment by regadas** [comment]
@JozoVilcek also is this a streaming pipeline or batch?

**Comment by JozoVilcek** [comment]
I am running batch pipeline

**Comment by regadas** [comment]
@JozoVilcek yeah this is an issue in `Beam` with `SparkPipelineResult` that does not follow `PipelineResult`.

as a quick test I ran our [WordCount](https://github.com/spotify/scio/blob/master/scio-examples/src/main/scala/com/spotify/scio/examples/WordCount.scala) with a big enough duration:

```
val sc: ScioContext = ???

sc.run().waitUntilFinish(1.hour)
```

So make sure you give a large duration for the time being.

**Comment by regadas** [comment]
The fix was merged in `Beam` not sure if it will be in `2.14.0`
